libmscore/edit.cpp

```diff
@@ -2252,12 +2252,15 @@ void Score::deleteMeasures(MeasureBase* is, MeasureBase* ie, bool preserveTies)
                         if (!s)
                               s = mAfterSel->undoGetSegment(SegmentType::TimeSig, mAfterSel->tick());
 
-                        TimeSig* nts = new TimeSig(this);
-                        nts->setTrack(staffIdx * VOICES);
-                        nts->setParent(s);
-                        nts->setFrom(lastDeletedForThisStaff);
-                        nts->setStretch(nts->sig() / mAfterSel->timesig());
-                        score->undoAddElement(nts);
+
+                        auto track = staffIdx * VOICES;
+                        if (track < s->elist().size()) {
+                              TimeSig* nts = new TimeSig(score);
```

Comment:
Two changes here:
1) verifying track is okay in relation to the score (in a loop going through all sub score or parts) (This gets verified in checkElement() function but it was asserting and crashing ungracefully
2) the new TimeSig is now getting the score for the part and not the this pointer which was the **master score**

Reply:
Seems the same change is needed for Mu4?

Reply:
Not sure, it's possible Mu4 has an appropriate check mechanism somewhere else down the line that would render this redundant...

Reply:
Now a 3rd change `setFrom(lastDeletedForThisStaff);` to `setSig(lastDeletedForThisStaff->sig());`